### PR TITLE
Add the Recyclarr app to DockStarter

### PR DESCRIPTION
**Pull request by SpiderManAPV**
Adds the Recyclarr app to dockstarter as an available option in the GUI menu.

# Pull request

**Purpose**
Adds the [Recyclarr](https://github.com/recyclarr/recyclarr) app as an option in DockStarter

**Approach**
Create a new folder in .apps for the Recyclarr container while following the [contributing guidelines](https://github.com/GhostWriters/DockSTARTer/blob/81a0e5c75eca52527478d8cef89719c8f597c9cd/.github/CONTRIBUTING.md)

**Requirements**
Check all boxes as they are completed

- [x] These changes meet the standards for [contributing](https://github.com/GhostWriters/DockSTARTer/blob/master/.github/CONTRIBUTING.md).
- [x] I have read the [code of conduct](https://github.com/GhostWriters/DockSTARTer/blob/master/.github/CODE_OF_CONDUCT.md).

### CODE REVIEW

File: compose/.apps/recyclarr/recyclarr.aarch64.yml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+services:
+  recyclarr:
+    image: ghcr.io/recyclarr/recyclarr
```
Comment (angrycuban13) [inline compose/.apps/recyclarr/recyclarr.aarch64.yml:3]: You will need a blank line at the end

File: compose/.apps/recyclarr/recyclarr.armv7l.yml [hunk 2]
```diff
@@ -0,0 +1,3 @@
+services:
+  recyclarr:
+    image: ghcr.io/recyclarr/recyclarr
```
Comment (angrycuban13) [inline compose/.apps/recyclarr/recyclarr.armv7l.yml:3]: You will need a blank line at the end

File: compose/.apps/recyclarr/recyclarr.hostname.yml [hunk 3]
```diff
@@ -0,0 +1,3 @@
+services:
+  recyclarr:
+    hostname: ${DOCKERHOSTNAME}
```
Comment (angrycuban13) [inline compose/.apps/recyclarr/recyclarr.hostname.yml:3]: You will need a blank line at the end

File: compose/.apps/recyclarr/recyclarr.labels.yml [hunk 4]
```diff
@@ -0,0 +1,10 @@
+services:
+  recyclarr:
+    labels:
+      com.dockstarter.appinfo.deprecated: "false"
+      com.dockstarter.appinfo.description: Application to synchronize settings from the TRaSH guides to your Sonarr/Radarr instances
+      com.dockstarter.appinfo.nicename: Recyclarr
+      com.dockstarter.appvars.recyclarr_enabled: "false"
+      com.dockstarter.appvars.recyclarr_network_mode: ""
+      com.dockstarter.appvars.recyclarr_restart: unless-stopped
+      com.dockstarter.appvars.recyclarr_create_config: "true"
```
Comment (angrycuban13) [inline compose/.apps/recyclarr/recyclarr.labels.yml:10]: You will need a blank line at the end

File: compose/.apps/recyclarr/recyclarr.netmode.yml [hunk 5]
```diff
@@ -0,0 +1,3 @@
+services:
+  recyclarr:
+    network_mode: ${RECYCLARR_NETWORK_MODE}
```
Comment (angrycuban13) [inline compose/.apps/recyclarr/recyclarr.netmode.yml:3]: You will need a blank line at the end

File: compose/.apps/recyclarr/recyclarr.x86_64.yml [hunk 6]
```diff
@@ -0,0 +1,3 @@
+services:
+  recyclarr:
+    image: ghcr.io/recyclarr/recyclarr
```
Comment (angrycuban13) [inline compose/.apps/recyclarr/recyclarr.x86_64.yml:3]: You will need a blank line at the end

File: compose/.apps/recyclarr/recyclarr.yml [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+    logging:
+      driver: json-file
+      options:
+        max-file: ${DOCKERLOGGING_MAXFILE}
+        max-size: ${DOCKERLOGGING_MAXSIZE}
+    restart: ${RECYCLARR_RESTART}
+    volumes:
+      - /etc/localtime:/etc/localtime:ro
+      - ${DOCKERSTORAGEDIR}:/storage
+      - ${DOCKERCONFDIR}/recyclarr:/config
```
Comment (angrycuban13) [inline compose/.apps/recyclarr/recyclarr.yml:18]: You will need a blank line at the end

File: docs/apps/recyclarr.md [hunk 8]
```diff
@@ -0,0 +1,15 @@
...
+
+## Description
+
+[Recyclarr](https://recyclarr.dev) Recyclarr is a command-line application that will automatically synchronize recommended settings from the TRaSH guides to your Sonarr/Radarr instances. It was formerly named "Trash Updater".
+
+## Install/Setup
+
+This application does not have any specific setup instructions documented. If
+you need assistance setting up this application please visit our
+[support page](https://dockstarter.com/basics/support/).
```
Comment (angrycuban13) [inline docs/apps/recyclarr.md:15]: You will need a blank line at the end

**Comment by angrycuban13** [comment]
You will also need to add `recyclarr.md` to the `mkdocs.yml` file

**Comment by SpiderManAPV** [comment]
Requested changes pushed to fork for PR.

File: .gitignore [hunk 9]
```diff
@@ -9,4 +9,4 @@ configs/
 dockstarter.log
 share/
 shared/
-site/
+site/
```
Comment (angrycuban13) [inline .gitignore:12]: You need a new line here as well since you modified the file

**Comment by angrycuban13** [comment]
@nemchik not sure what to make of this

**Review by nemchik — APPROVED** [review]

**Comment by nemchik** [comment]
> @nemchik not sure what to make of this
> 
> 

Fixed in https://github.com/GhostWriters/DockSTARTer/commit/c370f458e8593bb43236c8a26494a1a35f7d2901

Git started caring about the user that runs `git` commands. editorconfig-checker uses git to determine which files to check, but when it can't it falls back on finding all files. When it finds all files, it includes files in the `.git` folder, which don't match the `.editorconfig` rules. Running as the user (instead of as root) fixes the issue.

**Comment by Menz01** [comment]
would love to see this added without the need to do an override

**Comment by Menz01** [comment]
My bad. So sorry I looked last week and did not look more recently.

**Pull request merged**